From 82b5a827ac090b53aa4023b6a1473e38954ad125 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 7 Jan 2026 14:03:50 +0100
Subject: [PATCH 2/2] remoteproc: q6v5-wcss-sec: enable QDSS clock

IPQ6018 requires the QDSS clock to be enabled before booting the remoteproc
as otherwise the remoteproc will crash.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/remoteproc/qcom_q6v5_wcss_sec.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/remoteproc/qcom_q6v5_wcss_sec.c
+++ b/drivers/remoteproc/qcom_q6v5_wcss_sec.c
@@ -291,6 +291,7 @@ static int wcss_sec_probe(struct platfor
 	struct wcss_sec *wcss;
 	struct clk *sleep_clk;
 	struct clk *prng_clk;
+	struct clk *qdss_clk;
 	struct clk *int_clk;
 	const char *fw_name = NULL;
 	const struct wcss_data *desc = of_device_get_match_data(&pdev->dev);
@@ -331,6 +332,11 @@ static int wcss_sec_probe(struct platfor
 		return dev_err_probe(&pdev->dev, PTR_ERR(prng_clk),
 				     "Failed to get PRNG clock\n");
 
+	qdss_clk = devm_clk_get_optional_enabled(&pdev->dev, "qdss");
+	if (IS_ERR(qdss_clk))
+		return dev_err_probe(&pdev->dev, PTR_ERR(qdss_clk),
+				     "Failed to get PRNG clock\n");
+
 	ret = qcom_q6v5_init(&wcss->q6, pdev, rproc,
 			     WCSS_CRASH_REASON, NULL, NULL);
 	if (ret)
